pcsx2/GS/GSRegs.h

```diff
@@ -1156,32 +1174,80 @@ struct alignas(32) GIFPath
 
 		if (tag.FLG == GIF_FLG_PACKED)
 		{
-			if (regs.eq8(GSVector4i(0x0e0e0e0e)).mask() == (1 << nreg) - 1)
+			if (regs.eq8(GSVector4i::cxpr(0x0e0e0e0e)).mask() == (1 << nreg) - 1)
 			{
 				type = TYPE_ADONLY;
 			}
-			else
+			else if(nloop > 4)
 			{
 				switch (nreg)
 				{
 					case 1:
 						break;
 					case 2:
+						if (regs.U32[0] == 0x00000401)
```

Comment:
Could split this in two, and use select instead of branches, i.e. something like
```cpp
{
  const u32 val = regs.U32[0];
  if ((val >> 8) == 0x04)
  {
    type = (type == 0x401) ? TYPE_RGBAXYZF2 : ((type == 0x402) ? TYPE_UVXYZF2 : ((type == 0x403) ? TYPE_STQXYZF2 : type));
  }
  else if ((val >> 8) == 0x05)
  {
    type = (type == 0x501) ? TYPE_RGBAXYZ2 : ((type == 0x502) ? TYPE_STQXYZ2 : ((type == 0x503) ? TYPE_UVXYZ2: type));
  }
}
```
I _suspect_ selecting may be quicker than a lookup table, since you need to bounds check it anyway.